### PR TITLE
Retrieve subscribers of updated samples

### PR DESCRIPTION
**Pull request by jenniferboedker**
This PR implements the database connection and the listing of all subscribers whos subscribed projects have updated samples. The added use case SendNotification triggers the collection of updated samples for the day the tool is run.

The implementation of sending the and creating the message will be tackled in another PR.

Feel free to also give feedback to the structure of the tool in general, you might think using commanline parameters is usefull etc

### CODE REVIEW

**Comment by jenniferboedker** [comment]
[E02.02: Create a service that reads messages from the database and finds associated persons.](https://trello.com/c/dU3f4h8H/836-e0202-create-a-service-that-reads-messages-from-the-database-and-finds-associated-persons)

File: sample-notificator-domain/src/main/groovy/life/qbic/business/notification/send/SendNotification.groovy [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    private final SendNotificationDataSource ds
+
+    SendNotification(SendNotificationDataSource ds){
+        this.ds = ds
+    }
+
+    @Override
+    void sendNotifications() {
+        //todo when do we want to run the tool? Is "today" sufficient or do we want to do it for the day before?
+        List<Subscriber> subscribers = ds.getSubscribersForTodaysNotifications(Instant.now())
```
[the remark this answers is not in the document]
Comment (KochTobi) [reply sample-notificator-domain/src/main/groovy/life/qbic/business/notification/send/SendNotification.groovy:25]: https://www.concretepage.com/java/java-8/java-localdate-to-instant-timestamp

File: sample-notificator-domain/src/main/groovy/life/qbic/business/notification/send/SendNotificationDataSource.groovy [hunk 2]
```diff
@@ -9,4 +14,11 @@ package life.qbic.business.notification.send
  */
 interface SendNotificationDataSource {
 
+    /**
+     * Retrieves a list of subscribers, that need to be notified.
+     * Based on the current date (today) a list of subscribers that subscribed to an updated project will be returned.
+     * @param today The date of today
+     * @return a list of subscribers that need to be notified
+     */
+    List<Subscriber> getSubscribersForTodaysNotifications(Instant today) throws DatabaseQueryException
```
Comment (KochTobi) [inline sample-notificator-domain/src/main/groovy/life/qbic/business/notification/send/SendNotificationDataSource.groovy:23]: As far as I understood it, you need to handle it project-based and not subscriber based. Thus this method would need to specify the project you want to retrieve the subscribers for.

File: sample-notificator-domain/src/main/groovy/life/qbic/business/notification/send/SendNotificationOutput.groovy [hunk 3]
```diff
@@ -1,12 +0,0 @@
...
-/**
- * <b><short description></b>
- *
- * <p><detailed description></p>
- *
- * @since 1.0.0
- */
-interface SendNotificationOutput {
-
-}
```
Comment (KochTobi) [inline sample-notificator-domain/src/main/groovy/life/qbic/business/notification/send/SendNotificationOutput.groovy:12]: Why did you delete that? this is the interface that is to be used to communicate to the email sending part in the app layer.

File: sample-notificator-domain/src/main/groovy/life/qbic/business/subscription/Subscriber.groovy [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+ *
+ * <p>A subscriber is described by his full name and an email address</p>
+ *
+ * @since 1.0.0
+ *
+*/
+class Subscriber {
+    final String firstName
+    final String lastName
+    final String email
```
Comment (KochTobi) [inline sample-notificator-domain/src/main/groovy/life/qbic/business/subscription/Subscriber.groovy:16]: we need to know the project he/she subscribed to. Or better think about it the other way around. We could also create a Project data object that holds information on all the subscribers and the changes within this project

File: sample-notificator-domain/src/main/groovy/life/qbic/business/subscription/Subscriber.groovy [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    /**
+     * Mapping sample codes to its new sample status
+     */
+    final Map<String, Status> subscriptions
```
Comment (KochTobi) [inline sample-notificator-domain/src/main/groovy/life/qbic/business/subscription/Subscriber.groovy:20]: we do not subscribe to single samples but to projects

**Review by KochTobi — CHANGES_REQUESTED** [review]
There are a lot of things I marked. However, I think the major thing is the way subscriptions are attributed. I could not test the application yet and I am a bit sceptical as to it's function on the command line.

File: sample-notificator-domain/src/main/groovy/life/qbic/business/subscription/Subscriber.groovy (hunk 5 above)
Comment (KochTobi) [reply sample-notificator-domain/src/main/groovy/life/qbic/business/subscription/Subscriber.groovy:20]: image

File: sample-notificator-domain/src/main/groovy/life/qbic/business/notification/send/SendNotificationOutput.groovy (hunk 3 above)
Comment (jenniferboedker) [reply sample-notificator-domain/src/main/groovy/life/qbic/business/notification/send/SendNotificationOutput.groovy:12]: And what will be the output here?
The use case itself is sending the email. This is one use case for the whole app for now. I think it can be added when it is required.

File: sample-notificator-app/src/main/groovy/life/qbic/samplenotificator/NotificatorApp.groovy [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+        String sqlDatabase = Objects.requireNonNull(properties.get("mysql.db"), "Mysql database name missing.")
+
+        DatabaseSession.init(user, password, host, port, sqlDatabase)
+
+        sendNotificationsToSubscriber()
+    }
+
+    private static Properties getProperties(){
+        Properties properties = new Properties()
+        File propertiesFile = new File(NotificatorApp.class.getClassLoader().getResource('developer.properties').toURI())
```
[the remark this answers is not in the document]
Comment (jenniferboedker) [reply sample-notificator-app/src/main/groovy/life/qbic/samplenotificator/NotificatorApp.groovy:32]: I know.. I was not sure how to do it differently. But you are right, I can provide the properties file through the commandline options

File: sample-notificator-app/src/main/groovy/life/qbic/samplenotificator/NotificatorApp.groovy [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+
+        String user = Objects.requireNonNull(properties.get("mysql.user"), "Mysql user missing.")
+        String password = Objects.requireNonNull(properties.get("mysql.pass"), "Mysql password missing.")
+        String host = Objects.requireNonNull(properties.get("mysql.host"), "Mysql host missing.")
+        String port = Objects.requireNonNull(properties.get("mysql.port"), "Mysql port missing.")
+        String sqlDatabase = Objects.requireNonNull(properties.get("mysql.db"), "Mysql database name missing.")
+
+        DatabaseSession.init(user, password, host, port, sqlDatabase)
+
+        sendNotificationsToSubscriber()
```
[the remark this answers is not in the document]
Comment (jenniferboedker) [reply sample-notificator-app/src/main/groovy/life/qbic/samplenotificator/NotificatorApp.groovy:27]: IMO its a very small tool which does not need to be so full of classes and interfaces (which bloats the amount of files and scattering the code). But I will have a look at it again

File: sample-notificator-app/src/main/groovy/life/qbic/samplenotificator/datasource/SendNotificationDbConnector.groovy [hunk 8]
```diff
@@ -0,0 +1,139 @@
+package life.qbic.samplenotificator.datasource
+
+import groovy.util.logging.Log4j2
+import life.qbic.business.exception.DatabaseQueryException
+import life.qbic.business.notification.send.SendNotificationDataSource
+import life.qbic.business.subscription.Subscriber
+import life.qbic.datamodel.samples.Status
+import life.qbic.samplenotificator.datasource.database.ConnectionProvider
+import org.apache.commons.collections.map.HashedMap
```
[the remark this answers is not in the document]
Comment (jenniferboedker) [reply sample-notificator-app/src/main/groovy/life/qbic/samplenotificator/datasource/SendNotificationDbConnector.groovy:9]: hmm.. you mean the hashmap?

File: sample-notificator-domain/src/main/groovy/life/qbic/business/notification/send/SendNotification.groovy (hunk 1 above)
[the remark this answers is not in the document]
Comment (jenniferboedker) [reply sample-notificator-domain/src/main/groovy/life/qbic/business/notification/send/SendNotification.groovy:25]: ... and I thought I am going to make you happy with the Instant

File: sample-notificator-domain/src/main/groovy/life/qbic/business/notification/send/SendNotificationDataSource.groovy (hunk 2 above)
Comment (jenniferboedker) [reply sample-notificator-domain/src/main/groovy/life/qbic/business/notification/send/SendNotificationDataSource.groovy:23]: Really? AFAIK we will send notifications per person, so a notification should contain the collection of projects AND sample codes that were updated. So I think it makes sense to retrieve the subscribers that will be notified

File: sample-notificator-domain/src/main/groovy/life/qbic/business/subscription/Subscriber.groovy (hunk 4 above)
Comment (jenniferboedker) [reply sample-notificator-domain/src/main/groovy/life/qbic/business/subscription/Subscriber.groovy:16]: from the list of subscriptions you can refer the project. Its within the sample code

File: sample-notificator-domain/src/main/groovy/life/qbic/business/subscription/Subscriber.groovy (hunk 5 above)
Comment (jenniferboedker) [reply sample-notificator-domain/src/main/groovy/life/qbic/business/subscription/Subscriber.groovy:20]: true, but we need the information about the sample. And we can infer the project information from a sample.

File: sample-notificator-domain/src/main/groovy/life/qbic/business/notification/send/SendNotificationOutput.groovy (hunk 3 above)
Comment (KochTobi) [reply sample-notificator-domain/src/main/groovy/life/qbic/business/notification/send/SendNotificationOutput.groovy:12]: no the use case must not send the email since this requires 3rd party tools

File: sample-notificator-app/src/main/groovy/life/qbic/samplenotificator/datasource/SendNotificationDbConnector.groovy (hunk 8 above)
[the remark this answers is not in the document]
Comment (KochTobi) [reply sample-notificator-app/src/main/groovy/life/qbic/samplenotificator/datasource/SendNotificationDbConnector.groovy:9]: yes, why is it needed?

[the remark this answers is not in the document]
Comment (jenniferboedker) [reply sample-notificator-app/src/main/groovy/life/qbic/samplenotificator/datasource/SendNotificationDbConnector.groovy]: ahh now I see it. I wanted to have a hashmap not a hashedmap. I removed it

File: sample-notificator-app/src/main/groovy/life/qbic/samplenotificator/datasource/notification/create/CreateNotificationDbConnector.groovy [hunk 9]
```diff
@@ -0,0 +1,136 @@
...
+
+            throw new DatabaseQueryException(e.message)
+        }
+
+        return subscriberList
+    }
+
+    private Map<String, Status> getTodaysNotifications(LocalDate today){
+        //todo is this efficient?
+        String sqlQuery = SELECT_NOTIFICATIONS + " WHERE year(arrival_time) = ? AND month(arrival_time) = ? AND day(arrival_time) = ?"
```
Comment (KochTobi) [inline sample-notificator-app/src/main/groovy/life/qbic/samplenotificator/datasource/notification/create/CreateNotificationDbConnector.groovy:56]:
```suggestion
        String sqlQuery = SELECT_NOTIFICATIONS + " WHERE arrival_time BETWEEN ? AND ?"
```
and fill this with startOfDay and endOfDay

File: sample-notificator-domain/src/main/groovy/life/qbic/business/notification/create/CreateNotification.groovy [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+class CreateNotification implements CreateNotificationInput{
+    private final CreateNotificationDataSource ds
+
+    CreateNotification(CreateNotificationDataSource ds, CreateNotificationOutput output){
+        this.ds = ds
+    }
+
+    @Override
+    void createNotifications() {
+        //todo when do we want to run the tool? Is "today" sufficient or do we want to do it for the day before?
```
Comment (KochTobi) [inline sample-notificator-domain/src/main/groovy/life/qbic/business/notification/create/CreateNotification.groovy:24]: maybe leave that open if it is not decided yet. Otherwise, you make assumptions you might stumble over in the future.
You could change the interface to expect a `java.time.LocalDate`

File: sample-notificator-domain/src/main/groovy/life/qbic/business/notification/create/CreateNotification.groovy [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+        List<Subscriber> subscribers = ds.getSubscribersForTodaysNotifications(LocalDate.now().minusDays(1))
```
Comment (KochTobi) [inline sample-notificator-domain/src/main/groovy/life/qbic/business/notification/create/CreateNotification.groovy:25]: I would move the logic of getting all notifications and then all subscribers for all the projects in here instead of the database.

File: sample-notificator-domain/src/main/groovy/life/qbic/business/notification/create/CreateNotificationDataSource.groovy [hunk 12]
```diff
@@ -0,0 +1,24 @@
...
+ */
+interface CreateNotificationDataSource {
+
+    /**
+     * Retrieves a list of subscribers, that need to be notified.
+     * Based on the current date (today) a list of subscribers that subscribed to an updated project will be returned.
+     * @param today The date of today
+     * @return a list of subscribers that need to be notified
+     */
+    List<Subscriber> getSubscribersForTodaysNotifications(LocalDate today) throws DatabaseQueryException
```
Comment (KochTobi) [inline sample-notificator-domain/src/main/groovy/life/qbic/business/notification/create/CreateNotificationDataSource.groovy:23]: If you decide for a custom day you should rename this method.
```suggestion
    List<Subscriber> getSubscribersForNotificationsAt(LocalDate today) throws DatabaseQueryException
```

File: sample-notificator-domain/src/main/groovy/life/qbic/business/notification/create/CreateNotificationOutput.groovy [hunk 13]
```diff
@@ -0,0 +1,13 @@
+package life.qbic.business.notification.create
+
+/**
+ * <h1><short description></h1>
+ *
+ * <p><detailed description></p>
+ *
+ * @since 1.0.0
+ *
+*/
```
Comment (KochTobi) [inline sample-notificator-domain/src/main/groovy/life/qbic/business/notification/create/CreateNotificationOutput.groovy:10]: please provide javadoc

File: sample-notificator-domain/src/main/groovy/life/qbic/business/notification/send/SendNotificationOutput.groovy [hunk 14]
```diff
@@ -1,12 +1,12 @@
 package life.qbic.business.notification.send
-
 /**
- * <b><short description></b>
+ * <h1><short description></h1>
```
Comment (KochTobi) [inline sample-notificator-domain/src/main/groovy/life/qbic/business/notification/send/SendNotificationOutput.groovy:3]: I think `b` looks better when rendered 😄

**Review by KochTobi — CHANGES_REQUESTED** [review]
I have some small suggestions on how to make your code scream what it wants from the database 😄

File: sample-notificator-app/src/main/groovy/life/qbic/samplenotificator/DependencyManager.groovy [hunk 15]
```diff
@@ -0,0 +1,65 @@
...
+        properties = getProperties(commandLineParameters.pathToConfig)
+        initializeDependencies()
+    }
+
+    private void initializeDependencies(){
+        setupDatabase()
+        setupCreateNotification()
+    }
+
+    void sendNotifications(){
```
[the remark this answers is not in the document]
Comment (jenniferboedker) [reply sample-notificator-app/src/main/groovy/life/qbic/samplenotificator/DependencyManager.groovy:32]: This should act as a method to trigger the sending of the notifications. Kind of like a starting point

File: sample-notificator-app/src/main/groovy/life/qbic/samplenotificator/cli/CommandLineParser.groovy [hunk 16]
```diff
@@ -0,0 +1,40 @@
...
+
+
+        if (commandLineParameters.helpRequested) {
+            CommandLine.usage(new NotificatorCommandLineOptions(), System.out)
+            System.exit(0)
+        }
+
+        if (commandLineParameters.pathToConfig == null || commandLineParameters.pathToConfig.isEmpty()) {
+            log.info "You have to provide a config file."
+            println "You have to provide a config file."
```
[the remark this answers is not in the document]
Comment (jenniferboedker) [reply sample-notificator-app/src/main/groovy/life/qbic/samplenotificator/cli/CommandLineParser.groovy:34]:
```suggestion
            log.info "You have to provide a config file."
```

File: sample-notificator-domain/src/main/groovy/life/qbic/business/notification/create/CreateNotification.groovy (hunk 11 above)
Comment (jenniferboedker) [reply sample-notificator-domain/src/main/groovy/life/qbic/business/notification/create/CreateNotification.groovy:25]: I was already also thinking about that, will do

File: sample-notificator-domain/src/main/groovy/life/qbic/business/notification/send/SendNotificationOutput.groovy (hunk 14 above)
Comment (jenniferboedker) [reply sample-notificator-domain/src/main/groovy/life/qbic/business/notification/send/SendNotificationOutput.groovy:3]:
```suggestion
 * <b><short description></b>
```

File: sample-notificator-app/src/main/groovy/life/qbic/samplenotificator/DependencyManager.groovy (hunk 15 above)
[the remark this answers is not in the document]
Comment (jenniferboedker) [reply sample-notificator-app/src/main/groovy/life/qbic/samplenotificator/DependencyManager.groovy]: addressed

File: sample-notificator-app/src/main/groovy/life/qbic/samplenotificator/NotificatorApp.groovy [hunk 17]
```diff
@@ -0,0 +1,26 @@
...
+    NotificatorApp (NotificatorCommandLineOptions commandLineParameters){
+        this.commandLineParameters = commandLineParameters
+    }
+
+    @Override
+    void run() {
+        DependencyManager dependencyManager = new DependencyManager(commandLineParameters)
+        CreateNotificationInput createNotification = dependencyManager.getCreateNotification()
+        createNotification.createNotifications(commandLineParameters.date)
+    }
```
Comment (KochTobi) [inline sample-notificator-app/src/main/groovy/life/qbic/samplenotificator/NotificatorApp.groovy:25]: nice

File: sample-notificator-app/src/main/groovy/life/qbic/samplenotificator/DependencyManager.groovy [hunk 18]
```diff
@@ -0,0 +1,64 @@
...
+ * <h1>Sets up the use cases</h1>
+ *
+ * @since 1.0.0
+ *
+*/
+@Log4j2
+class DependencyManager {
+
+    private Properties properties
+    private CreateNotification createNotification
```
Comment (KochTobi) [inline sample-notificator-app/src/main/groovy/life/qbic/samplenotificator/DependencyManager.groovy:20]: You can set the type of the properties to the interface
```suggestion
    private Properties properties
    private CreateNotificationInput createNotification
```

File: sample-notificator-app/src/main/groovy/life/qbic/samplenotificator/SampleNotificatorEntryPoint.groovy [hunk 19]
```diff
@@ -0,0 +1,29 @@
...
+@Log4j2
+class SampleNotificatorEntryPoint {
+
+    public static void main(String[] args) {
+        try {
+            NotificatorCommandLineOptions commandLineParameters =
+                    CommandLineParser.parseAndVerifyCommandLineParameters(args)
+
+            NotificatorApp app = new NotificatorApp(commandLineParameters)
+            app.run()
```
Comment (KochTobi) [inline sample-notificator-app/src/main/groovy/life/qbic/samplenotificator/SampleNotificatorEntryPoint.groovy:24]: You could also use the [QBiC classes](https://github.com/qbicsoftware/core-utils-lib/tree/master/src/main/java/life/qbic/cli) here.

File: sample-notificator-app/src/main/groovy/life/qbic/samplenotificator/cli/CommandLineParser.groovy [hunk 20]
```diff
@@ -0,0 +1,39 @@
...
+/**
+ * <h1>Parses the command line</h1>
+ *
+ * <p>A parser for processing the commandline input.</p>
+ *
+ * @since 1.0.0
+ *
+*/
+@Log4j2
+class CommandLineParser {
```
Comment (KochTobi) [inline sample-notificator-app/src/main/groovy/life/qbic/samplenotificator/cli/CommandLineParser.groovy:15]: I am not sure if it is an idea to replace this with https://github.com/qbicsoftware/core-utils-lib/blob/master/src/main/java/life/qbic/cli/ToolExecutor.java

File: sample-notificator-app/src/main/groovy/life/qbic/samplenotificator/cli/NotificatorCommandLineOptions.groovy [hunk 21]
```diff
@@ -0,0 +1,35 @@
+package life.qbic.samplenotificator.cli
+
+import picocli.CommandLine
```
Comment (KochTobi) [inline sample-notificator-app/src/main/groovy/life/qbic/samplenotificator/cli/NotificatorCommandLineOptions.groovy:3]: Please use our wrapper instead.
https://github.com/qbicsoftware/core-utils-lib/blob/master/src/main/java/life/qbic/cli/AbstractCommand.java

File: sample-notificator-app/src/main/groovy/life/qbic/samplenotificator/datasource/notification/create/CreateNotificationDbConnector.groovy [hunk 22]
```diff
@@ -0,0 +1,139 @@
...
+        Instant endOfDay = day.plusDays(1).atStartOfDay().minusNanos(1).atZone(ZoneId.systemDefault()).toInstant()
+        String sqlQuery = SELECT_NOTIFICATIONS + " WHERE arrival_time BETWEEN ? AND ?"
+        Map<String, Status> foundNotifications = new HashMap<>()
+
+        Connection connection = connectionProvider.connect()
+
+        connection.withCloseable {
+            PreparedStatement preparedStatement = it.prepareStatement(sqlQuery)
+            preparedStatement.setTimestamp(1, Timestamp.from(startOfDay))
+            preparedStatement.setTimestamp(2, Timestamp.from(endOfDay))
```
Comment (KochTobi) [inline sample-notificator-app/src/main/groovy/life/qbic/samplenotificator/datasource/notification/create/CreateNotificationDbConnector.groovy:68]: nice !

File: sample-notificator-app/src/main/groovy/life/qbic/samplenotificator/datasource/notification/create/CreateNotificationDbConnector.groovy [hunk 23]
```diff
@@ -0,0 +1,139 @@
...
+            while (resultSet.next()) {
+                String sampleCode = resultSet.getString("sample_code")
+                Status status = Status.valueOf(resultSet.getString("sample_status"))
+                foundNotifications.put(sampleCode,status)
+            }
+        }
+        return foundNotifications
+    }
+
+    private Map<Integer,List<String>> getSubscriberIdForSamples(Map<String,Status> sampleToStatus){
```
Comment (KochTobi) [inline sample-notificator-app/src/main/groovy/life/qbic/samplenotificator/datasource/notification/create/CreateNotificationDbConnector.groovy:81]: What is the intention of this method? I do not understand it I think. Is this a join in groovy?

**Review by KochTobi — CHANGES_REQUESTED** [review]
Thank you for addressing my requests.
I see you use picocli atm. I would suggest using our wrapper classes for that. This makes your life a bit easier I think

**Comment by luiskuhn** [comment]
The implementation looks good to me 👍

File: sample-notificator-app/src/main/groovy/life/qbic/samplenotificator/datasource/notification/create/CreateNotificationDbConnector.groovy (hunk 23 above)
Comment (jenniferboedker) [reply sample-notificator-app/src/main/groovy/life/qbic/samplenotificator/datasource/notification/create/CreateNotificationDbConnector.groovy:81]: I think this is something that can be tackled by someone working on the use case? I think its too much to also work on that. I would opt for taking it as is and rework it

File: sample-notificator-domain/src/main/groovy/life/qbic/business/notification/send/SendNotificationInput.groovy [hunk 24]
```diff
@@ -1,12 +1,12 @@
 package life.qbic.business.notification.send
-
 /**
- * <b><short description></b>
+ * <h1><short description></h1>
```
Comment (KochTobi) [inline sample-notificator-domain/src/main/groovy/life/qbic/business/notification/send/SendNotificationInput.groovy:3]: why is this `h1` everywhere all of a sudden?

File: sample-notificator-app/src/main/groovy/life/qbic/samplenotificator/cli/CommandLineParser.groovy [hunk 25]
```diff
@@ -0,0 +1,39 @@
...
+
+        if (commandLineParameters.helpRequested) {
+            CommandLine.usage(new NotificatorCommandLineOptions(), System.out)
+            System.exit(0)
+        }
+
+        if (commandLineParameters.pathToConfig == null || commandLineParameters.pathToConfig.isEmpty()) {
+            log.info "You have to provide a config file."
+            System.exit(1)
+        }
```
Comment (KochTobi) [inline sample-notificator-app/src/main/groovy/life/qbic/samplenotificator/cli/CommandLineParser.groovy:35]: I think the QBiCToolExecutor does do this already. So no need for the CommandLine here.
You can use `getCommand()`

File: sample-notificator-app/src/main/groovy/life/qbic/samplenotificator/cli/NotificatorCommandLineOptions.groovy [hunk 26]
```diff
@@ -0,0 +1,30 @@
...
+*/
+@CommandLine.Command(
+        name = "SampleNotificator",
+        description = "A service to send notifications to subscribers to inform them about changes within their projects")
+class NotificatorCommandLineOptions extends AbstractCommand{
+
+    @CommandLine.Option(
+            names = ["-d", "--date"],
+            required = true,
+            description = "Date of the day, for which status update notifications should be send. Required format: yyyy-mm-dd")
```
Comment (KochTobi) [inline sample-notificator-app/src/main/groovy/life/qbic/samplenotificator/cli/NotificatorCommandLineOptions.groovy:22]:
```suggestion
            description = "Date of the day, for which status update notifications should be sent. Required format: yyyy-MM-dd")
```
` yyyy-mm-dd` is incorrect `mm` is for minutes and not months, I think.

**Review by KochTobi — APPROVED** [review]

File: sample-notificator-app/src/main/groovy/life/qbic/samplenotificator/cli/CommandLineParser.groovy (hunk 25 above)
Comment (jenniferboedker) [reply sample-notificator-app/src/main/groovy/life/qbic/samplenotificator/cli/CommandLineParser.groovy:35]: Thats true, did not remove it yet
